frontend/packages/console-shared/src/hooks/perspective-utils.ts

```diff
@@ -30,6 +31,7 @@ export type PerspectiveVisibility = {
 export type Perspective = {
   id: string;
   visibility: PerspectiveVisibility;
+  pinnedResources?: ExtensionK8sModel[];
 };
```

Comment:
When we create a specific type here we could be also sure that this doesn't use kind:

```suggestion
export type PerspectivePinnedResource = {
  group: string;
  version: string;
  resource: string;
}

export type Perspective = {
  id: string;
  visibility: PerspectiveVisibility;
  pinnedResources?: PerspectivePinnedResource[];
};
```